### PR TITLE
Optimize #971 get mean control

### PR DESCRIPTION
**Pull request by BeatrizPadrela**
### Linked issue

Should work alreayd for Hadamard and for the other data formats

### CODE REVIEW

File: Functions/xASL_quant_GetMeanControl.m
Comment (MichaelStritt) [inline Functions/xASL_quant_GetMeanControl.m:1]: Nice header 👍

**Comment by MichaelStritt** [comment]
On the first look I can't see any hard errors. If you say you already tested it on a few flavors then I'm fine, because we'll re-test everything for the release really soon anyway. Let's wait for Jan's feedback though 👍

File: Modules/SubModule_ASL/xASL_wrp_ResampleASL.m
Comment (MichaelStritt) [inline Modules/SubModule_ASL/xASL_wrp_ResampleASL.m:134]: Nice use of one function as the argument of another 👍

File: Functions/xASL_quant_GetControlLabelOrder.m
Comment (MichaelStritt) [inline Functions/xASL_quant_GetControlLabelOrder.m:26]: Maybe update the date while we're at it, otherwise perfect 👍

**Review by MichaelStritt — APPROVED** [review]
Let's wait for Jan's approval 👍

**Comment by jan-petr** [comment]
Some comments: NumberOfAverages might not be reliable for Hadamard, so we should really calculate the size of each Hadamard blockSize = MatrixSize*numberTEs

Then we also don't need to precalculate all the other sizes as we can then index the imASLTimeSeries directly with 1:blockSize:end

**Comment by BeatrizPadrela** [comment]
> Some comments: NumberOfAverages might not be reliable for Hadamard, so we should really calculate the size of each Hadamard blockSize = MatrixSize*numberTEs
> 
> Then we also don't need to precalculate all the other sizes as we can then index the imASLTimeSeries directly with 1:blockSize:end

Indeed! Looks great now

File: Functions/xASL_quant_GetControlLabelOrder.m
Comment (jan-petr) [reply Functions/xASL_quant_GetControlLabelOrder.m]: Done!

**Review by jan-petr — APPROVED** [review]
OK

**Pull request merged**
